server/core/region.go

```diff
@@ -622,6 +628,9 @@ func GenerateRegionGuideFunc(enableLog bool) RegionGuideFunc {
 					region.GetReplicationStatus().GetStateId() != origin.GetReplicationStatus().GetStateId()) {
 				saveCache = true
 			}
+			if !origin.IsFromHeartbeat() {
+				isNew = true
```

Comment:
Should we set `saveKV`, `saveCache`, `needSync` to `true` if it is from heartbeat?